language/move-lang/stdlib/modules/vector.move

```diff
@@ -56,4 +55,35 @@ module Vector {
         length(v) == 0
     }
 
+    // Return true if `e` is in the vector `v`
+    public contains<Element>(v: &T<Element>, e: &Element): bool {
+        let i = 0;
+        let len = length(v);
+        while (i < len) {
+            if (borrow(v, i) == e) return true;
+            i = i + 1;
+        };
+        false
+    }
+
+    // Remove the `i`th element E of the vector, shifting all subsequent elements
+    // It is O(n) and preserves ordering
+    public remove<Element>(v: &mut T<Element>, i: u64): Element {
+        let len = length(v);
+        // i out of bounds; abort
+        if (i >= len) abort 10;
```

Comment:
We should probably pick a convention for the VECTOR_INDEX_OUT_OF_BOUNDS error code + add it to vm/src/errors.rs at some point.